src/compiler/checker.ts

```diff
@@ -23139,6 +23153,10 @@ namespace ts {
             return result;
         }
 
+        function isNodeWithinFunction(node: Node, functionDeclaration: FunctionLike) {

```

Comment:
I would rename this to `isNodeWithinConstructor` and put the code from my comment above it in. (Or maybe inline like I have it above? I'm not sure which reads better.)